docs/CONTRIBUTING.md

```diff
@@ -48,38 +48,60 @@ We generally submit **major changes** via direct email to our [mail center](mail
 * **Description** Elaborate on the feature, why it is needed and the time frame in which you'd like it to be completed. If you have any suggestions or advice, share it with the team.
 Also, mockups are very helpful - so you can try to make a simple screenshot to help us understand the concept.
 
-Now tag it with `enhancement` and `1 - Ready` and other relevant tags from the following list: `backend`, `frontend`, `easy`, 
-
+Now tag it with `enhancement` and `1 - Ready` and other relevant tags from the following list: `backend`, `frontend`, `easy`,
 
 ## Coding Rules
 
 **Branch Names**
 
-When working on your own bug fix or feature, make the changes on your forked repository:
-
-* Bug Fix: assign yourself to the [open issue](https://github.com/hasadna/anyway/issues), tag it with `2 - Working`, create a new local branch and name it `XXXX-something` where XXXX is the number of the issue.
-
-* Feature: create an [enhancement issue](#feature-requests) to announce your intentions or assign yourself to existing issue, and name it `XXXX-something` where XXXX is the number of the issue.
-
-
+Branch naming convention is as following:
+
+```
+TYPE-ISSUE_ID-DESCRIPTION
+
+examples:
+feat-1247-add-vision-zero-widgets
+refactor-1242-move-rank-field-to-meta
+```
+When `TYPE` can be:
+* **feat** - is a new feature
+* **doc** - documentation only changes
+* **cicd** - changes related to CI/CD system
+* **fix** - a bug fix
+* **refactor** -  code change that neither fixes a bug nor adds a feature
+
+**All PRs must include commit message with the changes description!**
+
+Branching system:
+*master* - used for production <br>
+*dev* - anything else <br>
+
+**Working Procedure**
+
+A standard procedure for working on an issue would be to:
+1. Change to dev branch: `git checkout dev` and pull changes from upstream `git pull upstream dev`
+2. Create new branch from `dev` , like: `refactor-1242-move-rank-field-to-meta`
+3. Work on your new branch - commit - repeat
+4. Change to dev branch: `git checkout dev` and git pull at `dev`: `git pull upstream dev`
+5. Change to your branch: `git checkout dev` and merge: `git merge your-branch-name` and solve conflicts if they exist
+6. Push branch dev to **your forked directory** using `git push origin dev` and open PR to hasadna's `dev` branch.
+7. Get a code review approval / reject
+8. After approval, merge your PR
+9. Delete your branch after merge
 
 **Clean Code**
 
-See the following Python and JavaScript format guidelines and stick to them: 
-* [Airbnb JavaScript Guidelines] (https://github.com/airbnb/javascript)
-* [Python Style Guide] (https://www.python.org/dev/peps/pep-0008/)
+See the following Python and JavaScript format guidelines and stick to them:
+* [Airbnb JavaScript Guidelines](https://github.com/airbnb/javascript)
+* [Python Style Guide](https://www.python.org/dev/peps/pep-0008/)
```

Comment:
@atalyaalon what do you think about adding a linter/formatters to the project?

Reply:
@isavion we have pylint - check out our tests under .github/workflows

Reply:
@isavion we have pylint - check out our tests under .github, workflows